src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -2965,21 +2976,30 @@ private void processUpstreamJavaCompilation(List<UpstreamProject> upstreamProjec
                     if (initialCompile) {
                         skipRunningTests = true;
                     }
+
+                    // try recompiling failing project modules that are not dependent on the current
+                    // module (upstream of the current module)
+                    if (!project.disableDependencyCompile && recompileDependencies && !initialCompile) {
```

Comment:
Since this combination of 3 booleans is used several times throughout this class, should it be combined into a boolean method for convenience?  e.g. something like `shouldRecompileDependencies(ProjectModule project)` or some other appropriate name